packages/main/src/plugin/util/exec.spec.ts

```diff
@@ -0,0 +1,117 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { expect, describe, test, vi, beforeEach, afterEach } from 'vitest';
+import type { RunOptions, RunResult } from '@podman-desktop/api';
+import { getInstallationPath, macosExtraPath, runCliCommand } from './exec.js';
+import * as util from '../../util.js';
+import { CancellationTokenImpl } from '../cancellation-token.js';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+describe('runCliCommand', () => {
+  test('should run the command and resolve with the result', async () => {
+    const command = 'echo';
+    const args = ['Hello, World!'];
+    const options: RunOptions = {
+      env: {
+        NODE_ENV: 'test',
+      },
+      token: new CancellationTokenImpl(),
+    };
+
+    const result: RunResult = await runCliCommand(command, args, options);
+
+    expect(result).toBeDefined();
+    expect(result.exitCode).toBe(0);
+    expect(result.stdout).toContain('Hello, World!');
+    expect(result.stderr).toBe('');
+  });
+
+  test('should reject with an error when the command execution fails', async () => {
+    const command = 'nonexistent-command';
+    const args: any[] = [];
+    const options: RunOptions = {
+      token: new CancellationTokenImpl(),
+    };
+
+    await expect(runCliCommand(command, args, options)).rejects.toThrowError(/Failed to execute command/);
+  });
+
+  test('should reject with an error when the execution is cancelled', async () => {
+    const command = 'sleep';
+    const args = ['2'];
+    const cancellationToken = new CancellationTokenImpl();
+    const options: RunOptions = {
+      token: cancellationToken,
+    };
+    cancellationToken.cancel();
+
+    await expect(runCliCommand(command, args, options)).rejects.toThrowError(/Execution cancelled/);
```

Comment:
there is something strange here. It seems that the function is always returning a result and result contains an exit code
but if we get an exception, how do we check the exit code or stderr/stdout properties ?